flex_config/file_sources/toml_source.py

```diff
@@ -0,0 +1,42 @@
+from typing import Any, Dict, TextIO, cast
+
+from .file_source import FileSource
+
+
+class TOMLSource(FileSource):
+    """
+    A ConfigSource that loads values from a TOML file.
+
+    Warning:
+        In order to use this source you must have [toml](https://github.com/uiri/toml) installed either
+        directly or by including the "toml" extra when installing this package
+        (e.g. `poetry install flex-config -E toml`).
+
+    Args:
+        path (pathlib.Path): a [pathlib.Path](https://docs.python.org/3/library/pathlib.html#basic-use) to a YAML file
+            to load.
+        path_must_exist (bool): if true, the constructor will raise a RuntimeError if the provided path doesn't exist.
+
+    Example:
+        ```python
+        from pathlib import Path
+        from flex_config import TOMLSource, ConfigSchema, construct_config
+
+        my_toml_file = Path("path/to/my.toml")
+        toml_source = TOMLSource(my_toml_file)
+
+        class MyConfigSchema(ConfigSchema):
+            thing1: str
+            thing2: int
+
+        my_config = load_sources(config_schema=MyConfigSchema, sources=toml_source)
+        ```
+
+    """
+
+    def _load_file(self, file: TextIO) -> Dict[str, Any]:
+        """ Loads and parses the TOML file """
+        import toml
+
+        config_dict = toml.load(file)
+        return cast(Dict[str, Any], config_dict)
```

Comment:
With this as well as JSON/YAML it's possible that the top level passed is not a dict and is instead a list right? So perhaps we should be doing some runtime checking and producing a useful error instead of whatever ends up failing further up.

Different issue though